core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryBuilder.java

```diff
@@ -111,65 +146,295 @@ public GeoDistanceRangeQueryBuilder includeLower(boolean includeLower) {
         this.ignoreMalformed = ignoreMalformed;
         return this;
     }
 
+    public boolean ignoreMalformed() {
+        return ignoreMalformed;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException errors = new QueryValidationException();
+        if (fieldName == null) {
+            errors.addValidationError("fieldName cannot be null");
+        }
+        if (lat != null && lon == null || lon != null && lat == null) {
+            errors.addValidationError("Both lat and lon must be defined. Found lat [" + lat + "] and lon [" + lon + "]");
+        }
+        if ((lat == null && lon == null && geohash == null) || (geohash != null && (lat != null || lon != null))) {
+            errors.addValidationError("Must either define geohash or lat and lon fields. Found geohash [" + geohash + "], lat [" + lat
+                    + "] and lon [" + lon + "]");
+        }
+        if (from == null && to == null) {
+            errors.addValidationError("Must define at least one parameter from [from, to]");
+        }
+        if (from != null && !(from instanceof Number || from instanceof String)) {
+            errors.addValidationError("from must either be a number or a string. Found [" + from.getClass().getName() + "]");
+        }
+        if (to != null && !(to instanceof Number || to instanceof String)) {
+            errors.addValidationError("to must either be a number or a string. Found [" + to.getClass().getName() + "]");
+        }
+        if (optimizeBbox != null && !(optimizeBbox.equals("none") || optimizeBbox.equals("memory") || optimizeBbox.equals("indexed"))) {
+            errors.addValidationError("optimizeBbox must be one of [none, memory, indexed]");
+        }
+        return errors.validationErrors().isEmpty() ? null : errors;
+    }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        if (coerce) {
+            ignoreMalformed = true;
+        }
+
+        final boolean indexCreatedBeforeV2_0 = context.indexVersionCreated().before(Version.V_2_0_0);
+        GeoPoint point;
+        if (geohash == null) {
+            point = new GeoPoint(lat, lon);
+        } else {
+            point = new GeoPoint();
+            GeoHashUtils.decode(geohash, point);
+        }
+        // validation was not available prior to 2.x, so to support bwc
+        // percolation queries we only ignore_malformed on 2.x created indexes
+        if (!indexCreatedBeforeV2_0 && !ignoreMalformed) {
+            if (point.lat() > 90.0 || point.lat() < -90.0) {
+                throw new QueryParsingException(context.parseContext(), "illegal latitude value [{}] for [{}]", point.lat(), NAME);
+            }
+            if (point.lon() > 180.0 || point.lon() < -180) {
+                throw new QueryParsingException(context.parseContext(), "illegal longitude value [{}] for [{}]", point.lon(), NAME);
```

Comment:
these should be QueryShardException instead, given that we are not parsing anymore, but rather on the shard.